src/Compilers/Core/MSBuildTaskTests/CscTests.cs

```diff
@@ -27,6 +27,15 @@ public void MultipleSourceFiles()
             Assert.Equal("/out:test1.exe test1.cs test2.cs", csc.GenerateResponseFileContents());
         }
 
+        [Fact] 
+        public void DeterministicFlag()
+        {
+            var csc = new Csc();
+            csc.Sources = MSBuildUtil.CreateTaskItems("test.cs");
+            csc.Deterministic = true;
+            Assert.Equal("/deterministic+ /out:test.exe test.cs", csc.GenerateResponseFileContents());
+        }
```

Comment:
Can we get a negative test here?  Making sure that /deterministic- gets emitted when we use false?